util/resolver/resolver.go

```diff
@@ -192,3 +187,33 @@ func hostsWithCredentials(ctx context.Context, hosts docker.RegistryHosts, sm *s
 		return res, nil
 	}
 }
+
+func newDefaultClient() *http.Client {
+	return &http.Client{
+		Transport: newDefaultTransport(),
+	}
+}
+
+// newDefaultTransport is for pull or push client
+//
+// NOTE: For push, there must disable http2 for https because the flow control
+// will limit data transfer. The net/http package doesn't provide http2 tunable
+// settings which limits push performance.
```

Comment:
Is there just no way to force it to open up multiple connections? The flow control will actually make the number of bytes sent over the wire smaller than it would on a single connection or is there multiplexing overhead?